Modules/Filtering/DisplacementField/test/itkDisplacementFieldJacobianDeterminantFilterTest.cxx

```diff
@@ -118,7 +120,113 @@ TestDisplacementJacobianDeterminantValue()
   }
   else
   {
-    std::cout << "Test passed." << std::endl;
+    std::cout << "Determinant value test passed" << std::endl;
+  }
+
+  // Now test that the determinant is consistent if the voxel space is different, but the physical space is the same.
+
+  // Define physical point to test
+  itk::Point<double, 2> physPt;
+  dispacementfield->TransformIndexToPhysicalPoint(index, physPt);
+
+  // Use this function to get the determinant at a specified physical point (it will be a different index between tests)
+  auto GetDeterminantAtPoint = [](FieldType::Pointer image, const itk::Point<double, 2> & pt) -> float {
+    auto filter = FilterType::New();
+    filter->SetInput(image);
+    filter->SetUseImageSpacing(true);
+    filter->Update();
+
+    itk::Index<2> mappedIdx = image->TransformPhysicalPointToIndex(pt);
+
+    return filter->GetOutput()->GetPixel(mappedIdx);
+  };
+
+  const float detOriginal = GetDeterminantAtPoint(dispacementfield, physPt);
+
+  // Check that this is the same as above, just to be sure the function above works
+  if (itk::Math::abs(detOriginal - expectedJacobianDeterminant) > epsilon)
+  {
+    std::cerr.precision(static_cast<int>(itk::Math::abs(std::log10(epsilon))));
+    std::cerr << "Test failed " << std::endl;
+    std::cerr << "Error in pixel value at physical point " << physPt << std::endl;
+    std::cerr << "Expected value " << detOriginal << std::endl;
+    std::cerr << " differs from " << expectedJacobianDeterminant;
+    std::cerr << " by more than " << epsilon << std::endl;
+    testPassed = false;
+  }
+
+  using PermuteFilterType = itk::PermuteAxesImageFilter<FieldType>;
```

Comment:
I think this long-ish code of permutations can be replaced by a few invocations of `OrientImageFilter`, which updates direction according to permutations to retain same physical space.

Reply:
Hi, I tried to do this but it looks like OrientImageFilter requires 3D input.

I could change the test data to 3D but I think it would change the ground truth in the original test.